Assets/MRTK/SDK/Experimental/Elastic/Scripts/ElasticSystem.cs

```diff
@@ -0,0 +1,212 @@
+
+        /// <summary>
+        /// Query the elastic system for the current instantaneous value
+        /// </summary>
+        /// <returns>Current value of the elastic system</returns>
+        public T GetCurrentValue() => currentValue;
+
+        /// <summary>
+        /// Query the elastic system for the current instantaneous velocity
+        /// </summary>
+        /// <returns>Current value of the elastic system</returns>
+        public T GetCurrentVelocity() => currentVelocity;
+    }
+
+    internal class LinearElasticSystem : ElasticSystem<float>
+    {
+        public LinearElasticSystem(float initialValue, float initialVelocity,
+                                   ElasticExtentProperties<float> extentInfo,
+                                   ElasticProperties elasticProperties)
+                                   : base(initialValue, initialVelocity,
+                                          extentInfo, elasticProperties) { }
+
+        /// <summary>
+        /// Update the internal state of the damped harmonic oscillator, given the forcing/desired value.
+        /// </summary>
+        /// <param name="forcingValue">Forcing function, for example, a desired manipulation position.
+        /// See https://en.wikipedia.org/wiki/Forcing_function_(differential_equations). It is a non-time-dependent
+        /// input function to a differential equation; in our situation, it is the "input position" to the spring.</param>
+        /// <param name="deltaTime">Amount of time that has passed since the last update.</param>
+        public override float ComputeIteration(float forcingValue, float deltaTime)
+        {
+            // F = -kx - (drag * v)
+            var force = (forcingValue - currentValue) * elasticProperties.HandK - elasticProperties.Drag * currentVelocity;
+
+            // Distance that the current stretch value is from the end limit.
+            float distFromEnd = extentInfo.MaxStretch - currentValue;
+
+            // Add force to the "left" if we are beyond the max stretch.
+            // (or, if enabled, add snapping force if we are near the endpoint)
+            force -= computeEndForce(currentValue - extentInfo.MaxStretch);
+
+            // Add force to the "right" if we are beyond the max stretch.
+            // (or, if enabled, add snapping force if we are near the endpoint)
+            force += computeEndForce(extentInfo.MinStretch - currentValue);
+
+            // Iterate over each snapping point, and apply forces as necessary.
+            foreach (float snappingPoint in extentInfo.SnapPoints)
+            {
+                // Calculate distance from snapping point.
+                var distFromSnappingPoint = snappingPoint - currentValue;
+                force += computeSnapForce(distFromSnappingPoint, elasticProperties.SnapK, elasticProperties.SnapRadius);
+            }
+
+            // Helper closure to reduce force calculation copypasta.
+            float computeEndForce(float current)
```

Comment:
We don't have a lot of other usage of nested functions elsewhere - let's make these local private ones for now (I want to ensure we get some agreement first about where we like it etc)

Reply:
OK. Personally I'm an advocate for "private functions" when they serve a purpose as closures specifically. These were originally closures but I changed them into functions... but then didn't separate them.

Given that these aren't actually closures, it's separate them like you've suggested!